### DISCUSSION TITLE
support retry if failed

### DISCUSSION THREAD

**Discussion by shikbupt** [earthboundkid/requests#61] [Support] [answered]
Backoff Retry Mechanism can be added ?

**Comment by earthboundkid** [comment, marked as answer]
See https://github.com/carlmjohnson/requests/wiki#retries

**Reply by tmm1** [reply]
will this allow retries on handler errors? seems like it only acts on the transport layer

**Reply by earthboundkid** [reply]
Yes, those retry packages only work at the transport level, so you would retry for timeouts and 500s etc. If you need to retry based on the response body, it would have to be done at a higher level.

**Reply by tmm1** [reply]
> timeouts and 500s etc

hm, does this mean validator errors (i.e. via `CheckStatus`) would be retried?

EDIT: never mind,  i guess you'd use something like `rehttp.RetryStatuses`
